tools/app.hpp

```diff
@@ -33,16 +33,30 @@ namespace arg {
         std::string m_encoding;
     };
 
+    enum class WandMode : bool { Required, Optional };
+
+    template <WandMode Mode = WandMode::Required>
     struct WandData {
         explicit WandData(CLI::App* app)
         {
-            auto* wand = app->add_option("-w,--wand", m_wand_data_path, "WAND data filename");
-            app->add_flag("--compressed-wand", m_wand_compressed, "Compressed WAND data file")
-                ->needs(wand);
+            if constexpr (Mode == WandMode::Required) {
+                auto* wand =
+                    app->add_option("-w,--wand", m_wand_data_path, "WAND data filename")->required();
+                app->add_flag("--compressed-wand", m_wand_compressed, "Compressed WAND data file")
+                    ->needs(wand);
+            } else {
+                auto* wand = app->add_option("-w,--wand", m_wand_data_path, "WAND data filename");
+                app->add_flag("--compressed-wand", m_wand_compressed, "Compressed WAND data file")
+                    ->needs(wand);
```

Comment:
We can take inside of `else` out to the top of the function, and then only add
```cpp
if constexpr (Mode == WandMode::Required) {
    wand->required();
}
```
to avoid code duplication.